docs/common-github-actions-practices.md

```diff
@@ -0,0 +1,387 @@
+
+      - name: Autobuild
+        uses: github/codeql-action/autobuild@v1
+
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v1
+
+  issue:
+    # open an issue on failure because it can be easy to miss CI failure notifications
+    name: Open issue on failure
+    needs: analyze
+    if: failure()
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Create issue
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh issue create --title "$GITHUB_WORKFLOW #$GITHUB_RUN_NUMBER failed" \
+                          --label bug \
+                          --body "See [$GITHUB_WORKFLOW #$GITHUB_RUN_NUMBER](https://github.com/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID)."
+```
+
+## Additional checks
+
+### Automated check for markdown links
+
+https://github.com/tcort/markdown-link-check checks markdown files for valid links and anchors.
+
+You may wish to not make this a required check for PRs to avoid unnecessarily blocking of PRs if external links break.
+
+```
+  markdown-link-check:
+    # release branches are excluded to avoid unnecessary maintenance when external links break
+    if: ${{ !startsWith(github.ref_name, 'v') }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install markdown-link-check
+        run: npm install -g markdown-link-check
+
+      - name: Run markdown-link-check
+        run: |
+          find . -type f \
+                 -name '*.md' \
+                 -not -path './CHANGELOG.md' \
+                 -exec markdown-link-check --config .github/scripts/markdown-link-check-config.json {} \;
+```
+
+The file `.github/scripts/markdown-link-check-config.json` is for configuring the markdown link check:
+
+```
+{
+  "retryOn429": true
+}
+```
+
+`retryOn429` helps with GitHub throttling.
+
+### Automated check for misspellings
+
+https://github.com/client9/misspell only checks against known misspellings,
+so while it's not a comprehensive spell checker, it doesn't produce false positives,
+and so doesn't get in your way.
+
+You may wish to not make this a required check for PRs to avoid unnecessarily blocking of PRs if
+new misspellings are added to the misspell dictionary.
+
+```
+  misspell-check:
+    # release branches are excluded to avoid unnecessary maintenance when
+    # new misspellings are added to the misspell dictionary
+    if: ${{ !startsWith(github.ref_name, 'v') }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install misspell
+        run: |
+          curl -L -o ./install-misspell.sh https://git.io/misspell
+          sh ./install-misspell.sh
+
+      - name: Run misspell
+        run: bin/misspell -error .
+```
+
+### Markdown lint
+
+Specification repo uses https://github.com/DavidAnson/markdownlint.
+
+Go repo uses https://github.com/avto-dev/markdown-lint.
+
+TODO
+
+### Running checks against changed files only
```

Comment:
Do we have any examples of doing this in any of the three repos?

Reply:
no, the go repository does though, and I thought it would be cool for link checks which are slow and slightly flaky, but then realized the fallacy of incremental link checks 😂